File: net-core-apps/Monolito para uma promotora de crédito/Aplicacao/Model/Cliente/ClienteModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Aplicacao.Model.Cliente
{
    public class ClienteModel
    {
        [Required]
        public int IdGenero { get; set; }

        [Required]
        public int IdEstadoCivil { get; set; }

        [Required]
        public int IdGrauInstrucao { get; set; }

        [Required]
        public int IdCidadeNatal { get; set; }

        [Required]
        public string Nome { get; set; }

        [Required]
        public DateTime DataNascimento { get; set; }

        public string Filiacao1 { get; set; }

        public string Filiacao2 { get; set; }

        [Required]
        public bool DeficienteVisual { get; set; }

        public int? IdLoja { get; set; }
        public int? IdProfissao { get; set; }
    }
}

File: react-js/Micro-frontend para uma empresa promotora de crédito/packages/core/src/plataforma-cliente/cliente/cliente.model.ts
import { GeneroModel } from './generos-query';
import { EstadoCivilModel } from './estados-civil-query';
import { GrauInstrucaoModel } from './graus-instrucao-query';

import { MunicipioModel } from '../localizacao/municipios-query';
import { LojaModel } from '../loja';

export interface ClienteModel {
  idGenero: number;
  idEstadoCivil: number;
  idGrauInstrucao: number;
  idCidadeNatal: number;
  nome: string;
  dataNascimento: string;
  filiacao1: string;
  filiacao2: string;
  deficienteVisual: boolean;
  email?: string;
  idLoja?: number;
}

export interface ClienteExibicaoModel {
  id: number;
  nome: string;
  cpf: string;
  dataNascimento: string;
  filiacao1: string;
  filiacao2: string;
  deficienteVisual: boolean;
  email: string;
  genero?: GeneroModel;
  estadoCivil?: EstadoCivilModel;
  grauInstrucao?: GrauInstrucaoModel;
  cidadeNatal?: MunicipioModel;
  loja?: LojaModel;
}
